Dockerfile

```diff
@@ -1,11 +1,26 @@
 # Running from node
 from node:lts-alpine3.11
 
-run apk add python make g++ ffmpeg
+# Set time zone - default of America/Phoenix
+ARG TZ='America/Phoenix'
+ENV DEFAULT_TZ ${TZ}
+RUN apk add -U tzdata \
+	&& cp /usr/share/zoneinfo/${DEFAULT_TZ} /etc/localtime \
+	&& apk del tzdata
 
-COPY ./ ./
+# Install build depenacies
+RUN apk add make gcc g++
+
+# Install run depenacies
+RUN apk add python ffmpeg
 
+# Copy over and build app
+COPY ./ ./
 run npm install
 
+#Clean up build dependancies and apk cache
+RUN apk del make gcc g++ \
+	&& rm -rf /var/cache/apk/*
```

Comment:
Why are we using more than one run line? Most of this should probably be done on one run line and chain them to save us space and time for deployment.

Reply:
Well This run line needs to happen after the npm install I can chain the apk adds though. It wont really save too much in the way of space or deployment time.

Reply:
Pushed in change that squashes the runs into just 2 runs. One before build and one after. What you lose is a bit of the documentation and readability. What you gain... maybe half a second of build time if even? ... that being half a second only if the step wasnt cached... which the apk add steps are pretty much always going to be cached unless you added to them.  Point I am trying to get to here is that optimizing docker files is generally not a great use of time because the build steps are very infrequent and usually automated (which is something we should work on for autodeploys). The most important thing to optimize in a docker is space used since that is actually going to have an impact on the production environment, hence why I performed a cleanup on the build tools

Reply:
After thinking on it we could further reduce the number of RUN commands by doing the following
```
# Running from node
from node:lts-alpine3.11

# Set time zone - default of America/Phoenix
ARG TZ='America/Phoenix'
ENV DEFAULT_TZ ${TZ}

# Copy over app
COPY ./ ./

# Set timezone, install build and run depenacies
RUN apk add -U tzdata \
  && cp /usr/share/zoneinfo/${DEFAULT_TZ} /etc/localtime \
  && apk del tzdata \
  && apk add make gcc g++ \
  && apk add python ffmpeg \
  && npm install \
  && apk del make gcc g++ \
        && rm -rf /var/cache/apk/*

#Starting the file
CMD ["index.js"]
```

However what you lose here is the caching of the apk adds which makes it so it has to do all of the build steps every single time you build the image even though the apk adds are rarely going to change between builds. You save on space by sparing an intermediate image (which shouldnt be that big), but lose on that caching speed. IDK. Personally I would rather have the cache. especially when running builds to test on my machine. The end image size is pretty much the same  either way.